Src/WindowsServer/WindowsServer.Shared/Implementation/AppServiceEnvVarMonitor.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Utility to monitor the value of environment variables which may change 
+    /// during the run of an application. Checks the environment variables 
+    /// intermittently.
+    /// </summary>
+    internal static class AppServiceEnvVarMonitor
+    {
+        // Environment variables tracked by this monitor. (internal to allow tests to modify them)
+        internal static readonly Dictionary<string, string> CheckedValues = new Dictionary<string, string>()
+        {
+            { "WEBSITE_SITE_NAME", string.Empty },
+            { "WEBSITE_HOME_STAMPNAME", string.Empty },
+            { "WEBSITE_HOSTNAME", string.Empty }
+        };
+
+        // When is the next time we will allow a check to occur? (internal to allow tests to modify this to avoid waits)
+        internal static DateTime NextCheckTime = DateTime.MinValue;
+
+        // how often we allow the code to re-check the environment
+        private static readonly TimeSpan CheckInterval = TimeSpan.FromSeconds(30);
```

Comment:
Let user configure it?

Reply:
Could do, not sure how much value would be there. I propose we raise an issue and chat about it for the next release.

Reply:
Let's not overexpose API we are not certain in and may want to pull in the very next PR. We can make it configurable later if we stick with this solution, we cannot unmake public API once it's in stable release without the breaking change.